k8s/upgrades/0.9.0-1.0.0/jiva/verify_volume_upgrade.sh

```diff
@@ -0,0 +1,140 @@
+    is_upgrade_failed=1
+fi
+replica_version=$(kubectl get deployment "$r_deploy_name" -n "$ns" \
+        -o jsonpath='{.metadata.labels.openebs\.io/version}')
+if [ "$replica_version" != "$target_upgrade_version" ]; then
+    echo "Failed to upgrade Replica deployment $r_deploy_name labels"
+    is_upgrade_failed=1
+fi
+
+#Verifying the upgraded service versions
+controller_svc_version=$(kubectl get svc "$c_svc_name" -n "$ns" \
+        -o jsonpath='{.metadata.labels.openebs\.io/version}')
+if [ "$controller_svc_version" != "$target_upgrade_version" ] ; then
+    echo "Failed to upgrade target service $c_svc_name labels"
+    is_upgrade_failed=1
+fi
+
+#Verifying the upgraded deployment images
+controller_images=$(kubectl get deployment "$c_deploy_name" -n "$ns" \
+        -o jsonpath='{range .spec.template.spec.containers[*]}{@.image}?{end}')
+for image in $(echo "$controller_images" | tr "?" " "); do
+    image_version=$(echo "$image" | cut -d ':' -f 2) 
+    if [ "$image_version" != "$target_upgrade_version" ] ; then
+        echo "Failed to upgrade controller deployment $c_deploy_name images"
+        is_upgrade_failed=1 
+    fi
+done
+
+replica_images=$(kubectl get deployment "$r_deploy_name" -n "$ns" \
+        -o jsonpath='{range .spec.template.spec.containers[*]}{@.image}?{end}')
+for image in $(echo "$replica_images" | tr "?" " "); do 
+    image_version=$(echo "$image" | cut -d ':' -f 2)
+    if [ "$image_version" != "$target_upgrade_version" ] ; then
+        echo "Failed to upgrade Replica deployment $r_deploy_name images"
+        is_upgrade_failed=1
+    fi
+done
+
+#Verifying running status of controller and replica pods
+running_ctrl_pod_count=$(kubectl get pods -n "$ns" \
+-l openebs.io/controller=jiva-controller,openebs.io/persistent-volume="$pv" \
+--no-headers | wc -l)
+if [ "$running_ctrl_pod_count" != 1 ]; then
+    echo "Failed to upgrade controller pod not running"
+    is_upgrade_failed=1
+fi
+
+running_rep_pod_count=$(kubectl get pods -n "$ns" \
+-l openebs.io/replica=jiva-replica,openebs.io/persistent-volume="$pv" \
+--no-headers | wc -l)
+if [ "$running_rep_pod_count" != "$replication_factor" ]; then
+    echo "Failed to upgrade replica pods not running"
+    is_upgrade_failed=1 
+fi
+
+#Verifying registered replica count
+retry=0
+replica_count=0
+while [[ "$replica_count" != "$replication_factor" && $retry -lt 60 ]]
+do 
+    ctr_pod=$(kubectl get pod -n "$ns" \
+        -l openebs.io/persistent-volume="$pv",openebs.io/controller=jiva-controller \
+        -o jsonpath="{.items[*].metadata.name}" \
+    )
+    
+    replica_count=$(kubectl exec -it "$ctr_pod" -n "$ns" --container "$container_name" \
+        -- bash -c "curl -s http://localhost:9501/v1/volumes" \
+        | grep -oP '("replicaCount":)[0-9]' | cut -d ':' -f 2
+    )
+    
+    retry=$(( retry+1 ))
+    sleep 5
+done
+
+if [ "$replica_count" != "$replication_factor" ]; then
+    echo "Failed to upgrade jiva replicas not registered"
```

Comment:
message can be like
'Failed validation for registered replica count.. expected $replication_factor but only $replica_count are registered`
check other error message whether it can be improved, instead of saying 'Failed to upgrade' as the start of error.. it can be like 'Failed validation..'

Reply:
Done